### ISSUE TITLE
Multiple spaces in Collate cause error

### ISSUE THREAD

**Issue by GeoBosh** [r-lib/pkgdown#836]
In some circumstances entries in Collate separated by more than one space cause error. A minimal example package is at https://github.com/GeoBosh/reprexes/tree/master/pkgdownIssuespacesInCollate
and a session transcript (on Windows) is below. File DESCRIPTION contains the following line
```
Collate: 'a.R'  'b.R'
```
where there are two spaces between the file names. This gives a 'permission denied' error since apparently an attempt is made to `readLines()` from a directory. 

The site builds successfully if the above line is changed to 
```
Collate: 'a.R' 'b.R'
```
(one space between the file names).

Here is the transcript from the erroneous run:
```
> pkgdown::build_site()
|
Reading from: '.../reprexes/pkgdownIssuespacesInCollate'
Writing to:   '.../reprexes/pkgdownIssuespacesInCollate/docs'
|
-
-
-
Writing 'index.html'
-
Updating pkgdownIssuespacesInCollate documentation
Loading pkgdownIssuespacesInCollate
Skipping missing files: R
<Warning message:
simpleError in file(con, "r"): cannot open the connection>
In file(con, "r") :
  cannot open file '...\reprexes\pkgdownIssuespacesInCollate\R': Permission denied
Error in file(con, "r") : cannot open the connection
```
And here is the session information:
```
> sessionInfo()
R version 3.5.1 Patched (2018-10-03 r75399)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252 
[2] LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.19       knitr_1.20         xml2_1.2.0         magrittr_1.5      
 [5] roxygen2_6.1.0     MASS_7.3-50        debugme_1.1.0      R6_2.3.0          
 [9] rlang_0.2.2        stringr_1.3.1      tools_3.5.1        cli_1.0.1         
[13] htmltools_0.3.6    commonmark_1.6     yaml_2.2.0         digest_0.6.17     
[17] assertthat_0.2.0   rprojroot_1.3-2    tibble_1.4.2       pkgdown_1.1.0.9000
[21] crayon_1.3.4       processx_3.2.0     purrr_0.2.5        callr_3.0.0       
[25] base64enc_0.1-3    fs_1.2.6           ps_1.1.0           evaluate_0.11     
[29] memoise_1.1.0      rmarkdown_1.10     stringi_1.1.7      pillar_1.3.0      
[33] compiler_3.5.1     desc_1.2.0         backports_1.1.2   
```

**Comment by jayhesselberth** [comment]
Thanks for the reprex, I can't reproduce this error locally.

**Comment by GeoBosh** [comment]
Could it be Windows related? I also noticed that if I build the package the additional space disappears and the site builds successfully.

**Comment by jayhesselberth** [comment]
It could possible be a Windows issue; I'm using OS X.

pkgdown uses `desc` internally to read `DESCRIPTION` via `pkgdown:::read_desc()`. Can you try running that to see if there's an error?

**Comment by GeoBosh** [comment]
`pkgdown:::read_desc()` seems ok when called directly:
```
> a <- pkgdown:::read_desc()
> a
Package: pkgdownIssuespacesInCollate
Title: Demonstrates issue with multiple space in Collate
Version: 0.0.0.9000
Authors@R (parsed):
    * Georgi N. Boshnakov <georgi.boshnakov@gmail.com> [aut, cre]
Description: Demonstrates issue with multiple spaces in field Collate in
    file 'DESCRIPTION'.
License: GPL (>= 2)
Encoding: UTF-8
LazyData: true
RoxygenNote: 6.1.0
Collate:
    'a.R'
    'b.R'
```
But looking at the method that gets the contents of collate, its argument seems to imply different handling for OS'es:
```
> a$get_collate
function (which = c("main", "windows", "unix")) 
idesc_get_collate(self, private, match.arg(which))
<environment: 0x000000000a55b778>
```
When called with no argument  or "main" it works ok here (on Windows):
```
> a$get_collate()
[1] "a.R" "b.R"
> a$get_collate("main")
[1] "a.R" "b.R"
```
... but when called with "windows", it returns an empty character vector:
```
> a$get_collate("windows")
character(0)
```
If the internal code indeed makes this callwith argument "windows", this 
would explain the error - `paste0(path, character(0)` returns the R sudirectory and trying to `readlines()` from it gives the error.

**Comment by jayhesselberth** [comment]
So this sounds a buglet in `desc`?

**Comment by GeoBosh** [comment]
Maybe not. I looked at the code of desc but it seems that "windows" and
"linux" in this function are used for files that are for the corresponding
system only,
pkgdown hardly uses the windows argument. At least now it is not so
mysterious as to why an attempt to read /R happens. There are other places
where character(0) is returned.

Finding out how pkgdown gets the list of R files may be the key to this.
I wasn't able to find how pkgdown obtains the list of R files. It's
difficult to pinpoint the error since it happens in 'callr' but there
errors are catched and  it surfaces only at the last line where the object
inherits from try-error.

**Comment by GeoBosh** [comment]
I think that the problem comes from version 6.1.0 of roxygen2. When I installed the previous released version of roxygen2 (6.0.1) the problem disappeared.   Then reappeared when I installed 6.1.0 again. 

Running the reprex package, 6.1.0 I gives
```
> options(error=recover)
> roxygen2::roxygenize()
Error in file(con, "r") : cannot open the connection
In addition: Warning message:
In file(con, "r") :
  cannot open file 'c:\Users\mcbssgb2\Az\Rdevel\reprexes\pkgdownIssuespacesInCollate\R': Permission denied

Enter a frame number, or 0 to exit   

1: roxygen2::roxygenize()
2: parse_package(base_path, env = NULL, registry = registry, global_options = 
3: lapply(files, tokenize_file, registry = registry, global_options = global_o
4: FUN(X[[i]], ...)
5: read_lines(file)
6: base::readLines(path, n = n, encoding = "UTF-8", warn = FALSE)
7: file(con, "r")

Selection: 2
Called from: read_lines(file)
Browse[1]> files
[1] "c:\\Users\\mcbssgb2\\Az\\Rdevel\\reprexes\\pkgdownIssuespacesInCollate\\R\\a.R"
[2] "c:\\Users\\mcbssgb2\\Az\\Rdevel\\reprexes\\pkgdownIssuespacesInCollate\\R"     
[3] "c:\\Users\\mcbssgb2\\Az\\Rdevel\\reprexes\\pkgdownIssuespacesInCollate\\R\\b.R"
```
The second element of `files` generates the error.  Still inside `parse_package()` we get
```
Browse[1]> tmdesc <- read_pkg_description(path)
Browse[1]> tmdesc$Collate
[1] "'a.R'  'b.R'"
Browse[1]> nchar(tmdesc$Collate)
[1] 12 # confirming that there are 2 spaces between the filenames
```
In `package_files()` which produces the above, this command seems to be the culprit generating the empty 2nd element:
```
Browse[1]>  scan(text = tmdesc$Collate %||% "", what = "", sep = " ",
    quiet = TRUE) 
[1] "a.R" ""    "b.R"
```
I think that `sep=" "` sets the separator to exactly one space. Leaving t to the default "" doesn't produce redundant empty strings:
```
Browse[1]>  scan(text = tmdesc$Collate %||% "", what = "", sep = "",
    quiet = TRUE) 
 scan(text = tmdesc$Collate %||% "", what = "", sep = "",
+     quiet = TRUE) 
[1] "a.R" "b.R"
```

**Comment by GeoBosh** [comment]
OK, I now see this is the same as roxygen2 issue#790, which has been fixed on github.
